packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1027,27 +1027,51 @@ export class SnapController extends BaseController<
    * Results from this method should be efficiently serializable.
    *
    * @param origin - The origin requesting the snap.
-   * @param snapId - The id of the snap.
-   * @param version - The version of the snap to install.
+   * @param _snapId - The id of the snap.
+   * @param versionRange - The semver range of the snap to install.
    * @returns The resulting snap object, or an error if something went wrong.
    */
   private async processRequestedSnap(
     origin: string,
-    snapId: SnapId,
-    version: string,
+    _snapId: SnapId,
+    versionRange: string,
   ): Promise<ProcessSnapResult> {
-    const existingSnap = this.getTruncated(snapId);
-    // For devX we always re-install local snaps.
-    if (existingSnap && !snapId.startsWith(SnapIdPrefixes.local)) {
-      if (satisfiesSemver(existingSnap.version, version)) {
-        return existingSnap;
-      }
+    try {
+      this.validateSnapId(_snapId);
+    } catch (err) {
       return {
         error: ethErrors.rpc.invalidParams(
-          `Version mismatch with already installed snap. ${snapId}@${existingSnap.version} doesn't satisfy requested version ${version}`,
+          `"${_snapId}" is not a valid snap id.`,
         ),
       };
     }
+    const snapId = _snapId as ValidatedSnapId;
+
+    const existingSnap = this.getTruncated(snapId);
+    // For devX we always re-install local snaps.
+    if (existingSnap && getSnapPrefix(snapId) !== SnapIdPrefixes.local) {
+      if (satisfiesSemver(existingSnap.version, versionRange)) {
+        return existingSnap;
+      }
+
+      try {
+        const updateResult = await this.updateSnap(
```

Comment:
I think we should have an accompanying UI PR to add an update prompt for this?

Reply:
Yes, we should ask the user before attempting to update a snap.

Reply:
Added a request for approval in a new commit